android/src/main/java/com/swmansion/reanimated/keyboard/WindowsInsetsManager.java

```diff
@@ -14,10 +16,32 @@
 public class WindowsInsetsManager {
 
   private boolean mIsStatusBarTranslucent = false;
+
+  private int statusBarHeight = 0;
+  private int paddingTop = 0;
+
+  private int paddingBottom = 0;
   private final WeakReference<ReactApplicationContext> mReactContext;
   private final Keyboard mKeyboard;
   private final NotifyAboutKeyboardChangeFunction mNotifyAboutKeyboardChange;
 
+  private final ViewTreeObserver.OnDrawListener statusBarChangeListener =
+      new ViewTreeObserver.OnDrawListener() {
+        @Override
+        public void onDraw() {
+          View statusBar = getStatusBarView();
+          int currentStatusBarHeight = statusBar.getHeight();
+          boolean isStatusBarTranslucent = getWindow().getStatusBarColor() == Color.TRANSPARENT;
+
+          if (isStatusBarTranslucent != mIsStatusBarTranslucent
```

Comment:
This is fired few times during the changes so I don't want to do update multiple times (it blocks the UI).